hera_pspec/pspecbeam.py

```diff
@@ -0,0 +1,291 @@
+import numpy as np, pyuvdata, os, conversions
+from scipy import integrate
+from scipy.interpolate import interp1d
+
+conversion = conversions.Cosmo_Conversions()
+
+class PSpecBeam(object):
+
+    def __init__(self):
+        pass
+
+    def power_beam_int(self, stokes='I'):
+        pass
+
+    def power_beam_sq_int(self, stokes='I'):
+        pass
+
+    def compute_pspec_scalar(self, lower_freq, upper_freq, num_freqs, stokes='I', taper='none', little_h=True, num_steps=10000):
+        pass
+
+class PSpecBeamGauss(PSpecBeam):
+
+    def __init__(self, FWHM, beam_freqs):
+        """
+        Object to store a toy (frequency independent) Gaussian beam in a PspecBeam object
+
+        Parameters
+        ----------
+        FWHM: float, in radians
+                Full width half max of the beam
+
+        beam_freqs: float, array-like
+                Frequencies over which this Gaussian beam is to be created
+        """
+        self.FWHM = FWHM
+        self.beam_freqs = beam_freqs
+
+    def power_beam_int(self, stokes='I'):
+        """
+        Computes the integral of the beam over solid angle to give
+        a beam area (in str). Uses analytic formula that the answer
+        is 2 * pi * FWHM**2 / 8 ln 2.
+
+        Trivially this returns an array (i.e., a function of frequency),
+        but the results are frequency independent.
+
+        Parameters
+        ----------
+        stokes: str, optional
+                Which Stokes parameter's beam to compute the scalar for.
+                'I', 'Q', 'U', 'V', although currently only 'I' is implemented
+                Default: 'I'
+        Returns
+        -------
+        primary beam area: float, array-like
+        """
+        if stokes != 'I':
+            raise NotImplementedError()
```

Comment:
It'd be good to replace this with `NotImplementedError("Only stokes='I' is currently supported.")`

Reply:
Done!